components/datetime/src/pattern/runtime/pattern.rs

```diff
@@ -0,0 +1,45 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::super::{reference, PatternItem, TimeGranularity};
+use alloc::vec;
+use zerovec::ZeroVec;
+
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct Pattern<'data> {
+    #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    pub items: ZeroVec<'data, PatternItem>,
+    pub(crate) time_granularity: TimeGranularity,
+}
+
+impl<'data> From<&'data reference::Pattern> for Pattern<'data> {
+    fn from(input: &'data reference::Pattern) -> Self {
+        Self {
+            items: ZeroVec::clone_from_slice(&input.items),
+            time_granularity: input.time_granularity,
+        }
+    }
+}
```

Comment:
Nit: I realized that now that you use `clone_from_slice` in both of these impls, the `&'data` version does not make a lot of sense.  Please delete it.